components/dashboard/src/workspaces/CreateWorkspacePage.tsx

```diff
@@ -0,0 +1,346 @@
+    const user = useCurrentUser();
+    return startWithOptions || user?.additionalData?.isMigratedToTeamOnlyAttribution;
+};
+
+export function CreateWorkspacePage() {
+    const user = useCurrentUser();
+    const location = useLocation();
+    const props = StartWorkspaceOptions.parseSearchParams(location.search);
+
+    const [useLatestIde, setUseLatestIde] = useState<boolean | undefined>(
+        props.ideSettings?.useLatestVersion !== undefined
+            ? props.ideSettings.useLatestVersion
+            : !!user?.additionalData?.ideSettings?.useLatestVersion,
+    );
+    const [selectedIde, setSelectedIde] = useState(
+        props.ideSettings?.defaultIde !== undefined
+            ? props.ideSettings.defaultIde
+            : user?.additionalData?.ideSettings?.defaultIde,
+    );
+    const [selectedWsClass, setSelectedWsClass] = useState<string | undefined>(props.workspaceClass);
+    const [errorWsClass, setErrorWsClass] = useState<string | undefined>(undefined);
+    const [repo, setRepo] = useState<string | undefined>(location.hash.substring(1));
+    const onSelectEditorChange = useCallback(
+        (ide: string, useLatest: boolean) => {
+            setSelectedIde(ide);
+            setUseLatestIde(useLatest);
+        },
+        [setSelectedIde, setUseLatestIde],
+    );
+    const [errorIde, setErrorIde] = useState<string | undefined>(undefined);
+    const [creating, setCreating] = useState<boolean>(false);
+
+    const [createResult, setCreateResult] = useState<WorkspaceCreationResult | undefined>(undefined);
+    const [createError, setCreateError] = useState<StartWorkspaceError | undefined>(undefined);
+    const [selectAccountError, setSelectAccountError] = useState<SelectAccountPayload | undefined>(undefined);
+
+    const createWorkspace = useCallback(
+        async (options?: Omit<GitpodServer.CreateWorkspaceOptions, "contextUrl">) => {
+            // add options from search params
+            const opts = options || {};
+
+            if (!opts.workspaceClass) {
+                opts.workspaceClass = selectedWsClass;
+            }
+            if (!opts.ideSettings) {
+                opts.ideSettings = {
+                    defaultIde: selectedIde,
+                    useLatestVersion: useLatestIde,
+                };
+            }
+            if (!repo) {
+                return;
+            }
+
+            try {
+                setCreating(true);
+                const result = await getGitpodService().server.createWorkspace({
+                    contextUrl: repo,
+                    ...opts,
+                });
+
+                setCreateResult(result);
+            } catch (error) {
+                setCreateError(error);
+            } finally {
+                setCreating(false);
+            }
+        },
+        [repo, selectedIde, selectedWsClass, useLatestIde],
+    );
+
+    const onClickCreate = useCallback(() => createWorkspace(), [createWorkspace]);
```

Comment:
I think this is an unnecessary wrapper around `createWorkspace`, as it will change whenever `createWorkspace` changes, and just calls it. We could just pass `createWorkspace` below where we're passing in `onClickCreate`

Reply:
The signature doesn't fit the expected click event, because both accepot a different first argument.